### PR TITLE
docs/flannel/flannel-config.md: add a new section on setting ETCD_SSL_DIR variable

### PR DESCRIPTION
**Pull request by radhikapc**
Add a section on setting ETCD_SSL_DIR variable. ref: https://github.com/coreos/bugs/issues/1775#issuecomment-283243950

### CODE REVIEW

File: flannel/flannel-config.md [hunk 1]
```diff
@@ -147,6 +147,10 @@ The last step is to enable `flanneld.service` in the Ignition config:
 
 *Important*: Other units that will run in containers, including those scheduled via fleet, should include `Requires=flanneld.service`, `After=flanneld.service`, and `Restart=always|on-failure` directives. These directive are necessary because flanneld.service may fail due to etcd not being available yet. It will keep restarting and it is important for Docker based services to also keep trying until flannel is up.
 
+### Setting the environment variables
+
+If etcd requires SSL certificates, flannel will need those to communicate with etcd. This is achieved by passing the directory in which the certificates are stored to the flannel wrapper. To do so, set the environment variable, `ETCD_SSL_DIR` to `/etc/ssl` in `/etc/flannel/options.env`.
```
Comment (crawford) [inline flannel/flannel-config.md]: `/etc/flannel/options.env` isn't read by the flannel wrapper today (we need to fix that). I would instead suggest to add `Environment=ETCD_SSL_DIR` to a drop-in for flanneld.service.

Comment (crawford) [inline flannel/flannel-config.md]: I take it back, we shouldn't read from `/etc/flannel/options.env`. The only reason `/run/flannel/options.env` exists is for legacy reasons (working around docker).

File: flannel/flannel-config.md [hunk 2]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
...
+
+``` yaml
+systemd:
+  units:
+    - name: flanneld.service
+      dropins:
+        - name: 50-ssl.conf
+          contents: |
+            [Service]
+            Environment=ETCD_SSL_DIR=/etd/ssl
```
Comment (joshix) [inline flannel/flannel-config.md]: et_c_d

File: flannel/flannel-config.md [hunk 3]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
 
 *Important*: Other units that will run in containers, including those scheduled via fleet, should include `Requires=flanneld.service`, `After=flanneld.service`, and `Restart=always|on-failure` directives. These directive are necessary because flanneld.service may fail due to etcd not being available yet. It will keep restarting and it is important for Docker based services to also keep trying until flannel is up.
 
+### Specifying SSL certificates
+
+Flannel requires SSL certificates to communicate with a secure etcd cluster. By default, flannel looks for these certificates in `/etc/ssl/etcd`. To use different certificates, add `Environment=ETCD_SSL_DIR` to a drop-in file for `flanneld.service`.
+
+For example:
+
+``` yaml
```
Comment (crawford) [inline flannel/flannel-config.md]:
```
```container-linux-config
```

File: flannel/flannel-config.md [hunk 4]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
...
+coreos:
```
Comment (crawford) [inline flannel/flannel-config.md]:
```
systemd
```

File: flannel/flannel-config.md [hunk 5]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
...
+  units:
+    - name: flanneld.service
+      drop-ins:
```
Comment (crawford) [inline flannel/flannel-config.md]:
```
dropins
```

File: flannel/flannel-config.md [hunk 6]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
...
+        - name: 50-ssl.conf
+          content: |
```
Comment (crawford) [inline flannel/flannel-config.md]:
```
contents
```

File: flannel/flannel-config.md [hunk 7]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
...
+            [Service]
+            Environment=ETCD_SSL_DIR=/etcd/ssl
+
```
Comment (crawford) [inline flannel/flannel-config.md]: delete line

File: flannel/flannel-config.md [hunk 8]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
...
+``` container-linux-config
+systemd:
+  units:
+    - name: flanneld.service
+      dropins:
+        - name: 50-ssl.conf
+          contents: |
+            [Service]
+            Environment=ETCD_SSL_DIR=/etcd/ssl
```
Comment (crawford) [inline flannel/flannel-config.md]: /etc/ssl

File: flannel/flannel-config.md [hunk 9]
```diff
@@ -147,6 +147,24 @@ The last step is to enable `flanneld.service` in the Ignition config:
 
 *Important*: Other units that will run in containers, including those scheduled via fleet, should include `Requires=flanneld.service`, `After=flanneld.service`, and `Restart=always|on-failure` directives. These directive are necessary because flanneld.service may fail due to etcd not being available yet. It will keep restarting and it is important for Docker based services to also keep trying until flannel is up.
 
+### Specifying SSL certificates
+
+Flannel requires SSL certificates to communicate with a secure etcd cluster. By default, flannel looks for these certificates in `/etc/ssl/etcd`. To use different certificates, add `Environment=ETCD_SSL_DIR` to a drop-in file for `flanneld.service`.
+
+For example:
```
Comment (crawford) [inline flannel/flannel-config.md]: Can you elaborate a bit more on how to use the example? A lot of our users won't know immediately how to use this config snippet.

**Pull request merged**